### PR TITLE
Use Fyle's TPA Files APIs instead of AWS S3 directly

### PR DESCRIPTION
**Pull request by chatterjeeshekhar**
\

### CODE REVIEW

File: apps/data_fetcher/utils.py [hunk 1]
```diff
@@ -231,39 +171,64 @@ def send_email(from_email, to_email, subject, content):
...
-        content = render_to_string('email_body.html', {'link':presigned_url})
+        content = render_to_string('email_body.html', {'link':download_url})
         send_email(settings.SENDER_EMAIL_ID, email_to, subject, content)
+        print('Signed Download URL: ', download_url)
+
     except Exception as e:
         logger.error('Error while notifying user due to %s', e)
         raise
 
+def upload_file_to_aws(refresh_token, file_path):
```
Comment (Shwetabhk) [inline apps/data_fetcher/utils.py:207]: move this to FyleSDK connector
and call the connector like we did in QBO API

File: apps/data_fetcher/utils.py [hunk 2]
```diff
@@ -231,39 +171,64 @@ def send_email(from_email, to_email, subject, content):
...
+    """
+    Upload file to AWS S3 using FyleSDk
+    :param file_path: Location to the file
+    return: [file_id, file_url]
+    """
+    fyle_connection = FyleSDK(
+        settings.FYLE_BASE_URL, settings.FYLE_CLIENT_ID, 
+        settings.FYLE_CLIENT_SECRET, refresh_token
+        )
```
Comment (Shwetabhk) [inline apps/data_fetcher/utils.py:216]: fix indentation

File: apps/data_fetcher/utils.py (hunk 1 above)
Comment (chatterjeeshekhar) [reply apps/data_fetcher/utils.py:207]: But in `qbo_api` we were using the `FyleConnector` from the package itself. The backup app doesn't have that.

Comment (chatterjeeshekhar) [reply apps/data_fetcher/utils.py:207]: OH Got it.

File: apps/backups/utils.py [hunk 3]
```diff
@@ -144,17 +88,20 @@ def schedule_backup(request, backup):
...
-            object_id=backup.id,
-            payload={'backup_id': backup.id},
+            org_user_id=org_user_id,
             job_description='Fetch backup_id {0} for user: {1}'.format(
                 backup.id, request.user
-            ))
+            ),
+            object_id=backup.id,
+            payload={'backup_id': backup.id}
+            )
```
Comment (Shwetabhk) [inline apps/backups/utils.py:104]: fix indentation

File: apps/backups/utils.py [hunk 4]
```diff
@@ -144,17 +88,20 @@ def schedule_backup(request, backup):
...
         if created_job is None:
```
Comment (Shwetabhk) [inline apps/backups/utils.py:105]: this is not needed

File: apps/backups/utils.py [hunk 5]
```diff
@@ -164,5 +111,5 @@ def schedule_backup(request, backup):
         backup.save()
         return True
     except Exception as excp:
```
Comment (Shwetabhk) [inline apps/backups/utils.py:113]: `as excp` is not needed

File: apps/data_fetcher/utils.py [hunk 6]
```diff
@@ -58,67 +59,28 @@ def extract_employee_details(self):
...
-        object_name = fyle_org_id +'/'+ path.split('/')[2]
-        s3_client = boto3.client('s3', aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
-                                 aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY)
-        try:
-            s3_client.upload_file(file_name, settings.S3_BUCKET_NAME, object_name)
-        except ClientError as e:
-            logging.error('Error while uploading to s3 for oject %s. Error: %s', object_name, e)
-            raise
+        mime = magic.Magic(mime=True)
+        content_type = mime.from_file(file_path)
```
Comment (Shwetabhk) [inline apps/data_fetcher/utils.py:71]: can we restrict to zip file content type instead of using and external library?

File: apps/backups/models.py [hunk 7]
```diff
@@ -24,15 +24,14 @@ class Backups(models.Model):
...
     # we are now storing S3 object name in file_path
-    file_path = models.CharField(null=True, max_length=512,
-                                 help_text='Cloud storage URL for this backup')
     created_at = models.DateTimeField(auto_now_add=True, help_text='Created at datetime')
     modified_at = models.DateTimeField(auto_now=True, help_text='Updated at datetime')
+    fyle_file_id = models.CharField(max_length=255, help_text='File ID of file', null=True)
 
     def __str__(self):
-        return self.name
+        return self
```
Comment (Shwetabhk) [inline apps/backups/models.py:34]: Why is this needed?

File: apps/data_fetcher/utils.py (hunk 6 above)
Comment (Shwetabhk) [reply apps/data_fetcher/utils.py:71]: just remove the magic library

File: apps/backups/models.py (hunk 7 above)
Comment (chatterjeeshekhar) [reply apps/backups/models.py:34]: Otherwise, we aren't getting the backup.id needed to pass to our FyleJobs

File: apps/data_fetcher/utils.py (hunk 6 above)
Comment (chatterjeeshekhar) [reply apps/data_fetcher/utils.py:71]: If we only have zip then I will remove this.

File: apps/backups/models.py (hunk 7 above)
Comment (Shwetabhk) [reply apps/backups/models.py:34]: not following, you'll still be able to do `backup.id` no?

Comment (chatterjeeshekhar) [reply apps/backups/models.py:34]: `backup = Backups.objects.create(name=name, current_state=current_state,
                                    object_type=ObjectLookup[object_type],
                                    filters=filters, data_format=data_format,
                                    fyle_org_id=fyle_org_id, user=user,
                                    fyle_refresh_token=refresh_token
                                    )
    return backup`
